cmd/kubeadm/app/images/images.go

```diff
@@ -47,19 +47,22 @@ func GetCoreImage(image, repoPrefix, k8sVersion, overrideImage string) string {
 
 // GetAllImages returns a list of container images kubeadm expects to use on a control plane node
 func GetAllImages(cfg *kubeadmapi.MasterConfiguration) []string {
+	repoPrefix := cfg.GetControlPlaneImageRepository()
 	imgs := []string{}
-	imgs = append(imgs, GetCoreImage(constants.KubeAPIServer, cfg.ImageRepository, cfg.KubernetesVersion, cfg.UnifiedControlPlaneImage))
-	imgs = append(imgs, GetCoreImage(constants.KubeControllerManager, cfg.ImageRepository, cfg.KubernetesVersion, cfg.UnifiedControlPlaneImage))
-	imgs = append(imgs, GetCoreImage(constants.KubeScheduler, cfg.ImageRepository, cfg.KubernetesVersion, cfg.UnifiedControlPlaneImage))
-	imgs = append(imgs, fmt.Sprintf("%v/%v-%v:%v", cfg.ImageRepository, constants.KubeProxy, runtime.GOARCH, kubeadmutil.KubernetesVersionToImageTag(cfg.KubernetesVersion)))
+	imgs = append(imgs, GetCoreImage(constants.KubeAPIServer, repoPrefix, cfg.KubernetesVersion, cfg.UnifiedControlPlaneImage))
+	imgs = append(imgs, GetCoreImage(constants.KubeControllerManager, repoPrefix, cfg.KubernetesVersion, cfg.UnifiedControlPlaneImage))
+	imgs = append(imgs, GetCoreImage(constants.KubeScheduler, repoPrefix, cfg.KubernetesVersion, cfg.UnifiedControlPlaneImage))
+	imgs = append(imgs, fmt.Sprintf("%v/%v-%v:%v", repoPrefix, constants.KubeProxy, runtime.GOARCH, kubeadmutil.KubernetesVersionToImageTag(cfg.KubernetesVersion)))
+
+	// pause and etcd are not available on the ci image repository so use the default image repository.
 	imgs = append(imgs, fmt.Sprintf("%v/pause-%v:%v", cfg.ImageRepository, runtime.GOARCH, "3.1"))
 
 	// if etcd is not external then add the image as it will be required
 	if cfg.Etcd.Local != nil {
 		imgs = append(imgs, GetCoreImage(constants.Etcd, cfg.ImageRepository, cfg.KubernetesVersion, cfg.Etcd.Local.Image))
 	}
 
-	dnsImage := fmt.Sprintf("%v/k8s-dns-kube-dns-%v:%v", cfg.ImageRepository, runtime.GOARCH, constants.KubeDNSVersion)
+	dnsImage := fmt.Sprintf("%v/k8s-dns-kube-dns-%v:%v", repoPrefix, runtime.GOARCH, constants.KubeDNSVersion)
```

Comment:
Not sure if we put this image in the same spot as the other images...

Reply:
It did not complain when I tested it manually (which is how I found that etcd and pause are not put there). So I think it's ok? 😬

Reply:
your call .. my risk threshold is high (or is it low) as we are in code freeze mode :)

Reply:
double checked, you are right. Pulling coredns was fine but k8s-dns was not.

Where can we get this image from?

Reply:
I figured it out. You were totally right, thanks for pointing this out.